### PR TITLE
Validate position values are unique for action parameters

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
Fixes: https://github.com/StackStorm/st2/issues/3317

Non-unique positions:

```
2017-06-13 12:42:50,387 WARNING [-] Failed to register actions: Failed to register action "/opt/stackstorm/packs/examples/actions/position_test.yaml" from pack "examples": Parameters random and max_pause have same position 0. Position values have to be unique.
Traceback (most recent call last):
  File "./st2common/bin/st2-register-content", line 22, in <module>
    sys.exit(content_loader.main(sys.argv[1:]))
  File "/mnt/src/storm/st2/st2common/st2common/content/bootstrap.py", line 387, in main
    register_content()
  File "/mnt/src/storm/st2/st2common/st2common/content/bootstrap.py", line 338, in register_content
    register_actions()
  File "/mnt/src/storm/st2/st2common/st2common/content/bootstrap.py", line 219, in register_actions
    raise e
ValueError: Failed to register action "/opt/stackstorm/packs/examples/actions/position_test.yaml" from pack "examples": Parameters random and max_pause have same position 0. Position values have to be unique.
```

Non-contiguous positions:

```
2017-06-13 12:46:42,948 WARNING [-] Failed to register actions: Failed to register action "/opt/stackstorm/packs/examples/actions/position_test.yaml" from pack "examples": Positions supplied [0, 3] for parameters are not contiguous.
Traceback (most recent call last):
  File "./st2common/bin/st2-register-content", line 22, in <module>
    sys.exit(content_loader.main(sys.argv[1:]))
  File "/mnt/src/storm/st2/st2common/st2common/content/bootstrap.py", line 387, in main
    register_content()
  File "/mnt/src/storm/st2/st2common/st2common/content/bootstrap.py", line 338, in register_content
    register_actions()
  File "/mnt/src/storm/st2/st2common/st2common/content/bootstrap.py", line 219, in register_actions
    raise e
ValueError: Failed to register action "/opt/stackstorm/packs/examples/actions/position_test.yaml" from pack "examples": Positions supplied [0, 3] for parameters are not contiguous.
```

### CODE REVIEW

**Comment by LindsayHill** [comment]
Looks good. Messages seem clear enough.

**Comment by bigmstone** [comment]
Can you get rid of the trace just before the error message and just print the error message a hundred times....this way it won't confuse anyone. :trollface:

**Comment by Kami** [comment]
Nice work, additional validation for user-friendliness, etc. is always good.

File: st2common/st2common/validators/api/action.py [hunk 1]
```diff
@@ -69,6 +70,16 @@ def _validate_parameters(action_ref, action_params=None, runner_params=None):
                     action_ref, action_param, action_param_attr,
                     value, runner_params[action_param].get(action_param_attr))
 
+        if 'position' in action_param_meta:
```
Comment (Kami) [inline st2common/st2common/validators/api/action.py]: Minor style thing - I would do something like this, but yours is also fine:

```python
position = action_param_meta.get('position', None)
param = position_params.get(position, None)

if position is not None and param is not None:
....
```

**Comment by Kami** [comment]
LGTM, but let's wait with merging till v2.3.0 is out.

**Comment by Kami** [comment]
It looks like this is done so I will go ahead and merge it into master so it can be included in v2.3.1.

**Pull request merged**